### PR TITLE
Enhance Authentication

### PR DESCRIPTION
**Pull request by stevekroh**
- Validate both the login and register screens
- Redirect the user to a "thank you" screen after registration
- Add the ability to log out
- Only scroll the main content area when necessary
- Change the top-most elements to divs rather than Paper
- Make travis reject format and lint errors.
- Format and de-lint the whole project

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*global module*/
 module.exports = function( config ) {
   config.set( {
-    frameworks: [ 'browserify', 'mocha', 'es6-shim' ],
+    frameworks: [ 'browserify', 'jquery-2.1.0', 'mocha', 'es6-shim' ],
```
Comment (stevekroh) [inline karma.conf.js]: Add jquery so we can make css assertions

File: www/index.html [hunk 2]
```diff
@@ -33,10 +33,11 @@
         <link rel="stylesheet" type="text/css" href="css/index.css">
         <link rel="stylesheet" type="text/css" href="css/mod.css">
         <link rel="stylesheet" type="text/css" href="css/flex-objects.css">
+        <link rel="stylesheet" type="text/css" href="css/entry.css">
+        <link rel="stylesheet" type="text/css" href="css/page.css">
 
         <!-- MDL resources -->
         <link rel="stylesheet" href="lib/mdl/material.css">
```
Comment (stevekroh) [inline www/index.html]: We eventually need to get rid of material.css, but it affects our fonts right now.

**Comment by stevekroh** [comment]
When looking over this PR, the authentication changes are mainly those in pure green. One important change is that React component includes are wrapped in `eslint` ignore comments. This is because the linter cannot detect the usage of React component variables, as those are expanded by the JSX transpiler.

File: www/js/app.js [hunk 3]
```diff
@@ -27,75 +31,71 @@ import PeekPointCard from './components/peek-point-card';
...
-          <Route path="add-point" component={AddPointPage}>
-            <IndexRoute component={AddPointLocation} />
-            <Route path="name" component={AddPointName} />
-            <Route path="description" component={AddPointDescription} />
-            <Route path="hours" component={AddPointHours} />
-            <Route path="amenities" component={AddPointAmenities} />
+          <Route path="settings" component={ SettingsPage } />
+          <Route path="login" component={ LoginPage } />
+          <Route path="register" component={ RegisterPage } />
+          <Route path="/logout" component={ LogoutPage } />
```
Comment (JRJurman) [inline www/js/app.js]: do we need the `/` in the start of the path?

File: www/js/components/device-storage.js [hunk 4]
```diff
@@ -1,43 +1,49 @@
-import React, {Component} from 'react';
+/*global cordova*/
```
Comment (JRJurman) [inline www/js/components/device-storage.js]: what is this comment for?

File: www/js/components/peek-point-card.js [hunk 5]
```diff
@@ -1,30 +1,34 @@
...
-        (season.seasonStart.month <= currentMonth <= season.seasonEnd.month) &&
-        (season.seasonStart.date <= currentDate <= season.seasonEnd.date))
-      {
+  getDays( seasons ) {
+    let seasonDays = seasons[ 0 ].days;
+    const date = ( new Date() );
+    const curMonth = date.getMonth();
+    const curDate = date.getDate();
+    seasons.forEach( season => {
+      /*esfmt-ignore start*/
```
Comment (JRJurman) [inline www/js/components/peek-point-card.js]: why are we doing an esfmt-ignore here?

File: www/js/app.js (hunk 3 above)
Comment (stevekroh) [inline www/js/app.js]: Nope, will fix that.

File: www/js/components/device-storage.js (hunk 4 above)
Comment (stevekroh) [inline www/js/components/device-storage.js]: The linter has a no-unused-variable rule. You have to tell the linter you are using globals.

File: www/js/components/peek-point-card.js (hunk 5 above)
Comment (stevekroh) [inline www/js/components/peek-point-card.js]: In order to make the if statement readable, I had to use custom formatting. This isn't normally the case, but the formatter didn't do a good job with the complex if.

File: www/js/containers/filter-page.js [hunk 6]
```diff
@@ -1,159 +1,138 @@
...
-    const { activeFilters, openServices, hideAlert } = this.props.filters;
+  constructor( props ) {
+    super( props );
+    const {activeFilters, openServices, hideAlert} = this.props.filters;
 
-    const filters = types.filter(type =>{
-      return !activeFilters.includes(type);
-    }); // copy by value
+    const filters = types.filter( type => {
+      return !activeFilters.indexOf( type ) >= 0;
```
Comment (stevekroh) [inline www/js/containers/filter-page.js]: We can do that, or better yet, lodash/contains(activeFilters, type).

File: www/js/components/peek-point-card.js (hunk 5 above)
Comment (stevekroh) [inline www/js/components/peek-point-card.js]: Actually, I looked into this again. We don't need the ignores here, and in rating-point-card.js

File: www/js/reducers/points.js [hunk 7]
```diff
@@ -14,43 +14,44 @@ export const SYNC_RECEIVE = 'pannier/points/SYNC_RECEIVE';
 export const SYNC_DELETE = 'pannier/points/DELETE';
 export const RELOAD = 'pannier/points/RELOAD';
 
-export default function reducer(state = [], action) {
-  switch(action.type) {
+function helpSyncReceive( state, action ) {
+  const idx = findIndex( state, point => point._id === action.id );
```
Comment (JRJurman) [inline www/js/reducers/points.js]: possibly fix variable name to something more meaningful?

**Comment by ethanjurman** [comment]
:+1: :+1:

**Pull request merged**